### PR TITLE
Remove unused TaggedVariable node

### LINKED ISSUE #584: Remove TaggedVariable

### PR DESCRIPTION
**Pull request by aslesarenko**
Closes #584 
This removes the unused TaggedVariableNode class and the related code.
The node was replaced by ValUse node.

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/serialization/OpCodes.scala [hunk 1]
```diff
@@ -68,8 +68,7 @@ object OpCodes extends ValueCodes {
   val Undefined: OpCode = OpCode @@ 0.toByte
 
   // variables
-  val TaggedVariableCode: OpCode = newOpCode(1)
-  val ValUseCode: OpCode = newOpCode(2)
+  val ValUseCode: OpCode = newOpCode(2)  // code 1 was used by TaggedVariable (now it is free)
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/serialization/OpCodes.scala:71]: Please describe how removing an opcode is not a consensus-critical issue. What if a new node will get this opcode?

File: sigmastate/src/main/scala/sigmastate/Values.scala [hunk 2]
```diff
@@ -385,20 +368,13 @@ object Values {
...
-  type TaggedBigInt = TaggedVariable[SBigInt.type]
-  type TaggedBox = TaggedVariable[SBox.type]
-  type TaggedGroupElement = TaggedVariable[SGroupElement.type]
-  type TaggedSigmaProp = TaggedVariable[SSigmaProp.type]
-  type TaggedAvlTree = TaggedVariable[SAvlTree.type]
-  type TaggedByteArray = TaggedVariable[SCollection[SByte.type]]
-
-  def TaggedBox(id: Byte): Value[SBox.type] = mkTaggedVariable(id, SBox)
-  def TaggedAvlTree(id: Byte): Value[SAvlTree.type] = mkTaggedVariable(id, SAvlTree)
+  type ValUseInt = ValUse[SInt.type]
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/Values.scala:371]: ScalaDoc, better with references to the spec on what ValUse is at all.

File: sigmastate/src/main/scala/sigmastate/eval/RuntimeCosting.scala [hunk 3]
```diff
@@ -982,7 +982,7 @@ trait RuntimeCosting extends CostingRules { IR: IRContext =>
   import sigmastate._
 
   protected def isOperationNode(v: SValue): Boolean = v match {
-    case _: Block | _: BlockValue | _: TaggedVariableNode[_] | _: ValNode | _: ValDef | _: ValUse[_] | _: FuncValue => false
+    case _: Block | _: BlockValue | _: ValNode | _: ValDef | _: ValUse[_] | _: FuncValue => false
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/eval/RuntimeCosting.scala:985]: How this is not consensus-critical?

**Comment by kushti** [comment]
Please provide definitions for ValUse, TaggedVariable (maybe references to the spec, also it seems consensus-critical changes could be made to the serialization layer.

**Comment by aslesarenko** [comment]
This is obsolete, a new PR will be necessary.

**Pull request closed without merging**